### ISSUE TITLE
Generation of scenes mixing via API in high traffic

### ISSUE THREAD

**Issue by marcelo-earth** [marcelo-earth/generative-manim#23]
In some cases, the scenes returned by the API can be mixed when there are multiple requests.

**Example:**

<img width="1048" alt="image" src="https://github.com/360macky/generative-manim/assets/47821093/995cb406-57f5-4cf9-9e83-fe81ebc38af5">

**Code:**

```py
class GraphArrow(Scene):
    def construct(self):
        # Create axes and labels
        axes = Axes(x_range=[0, 10], y_range=[0, 10])
        x_label = MathTex(r'\text{Espaço}').scale(0.575)
        y_label = MathTex(r'\text{Tempo}').scale(0.75)
        labels = VGroup(x_label.next_to(axes.x_axis, RIGHT), y_label.next_to(axes.y_axis, UP))

        # Create dots
        d = Dot(color=WHITE, radius=0.1)
        d.move_to([0, 0, 0])
        d2 = Dot(color=WHITE, radius=0.1)
        d2.move_to([-1, 0, 0])

        # Group dots
        g = VGroup(d, d2)
        g.arrange()

        # Create traced paths
        trace = VMobject()
        trace2 = VMobject()

        trace_points = []
        trace2_points = []

        def update_trace(trace):
            trace_points.append(d.get_center())
            if len(trace_points) > 1:
                new_path = DashedVMobject(Line(trace_points[-2], trace_points[-1]), num_dashes=1, equal_lengths=True)
                trace.add(new_path)

        def update_trace2(trace2):
            trace2_points.append(d2.get_center())
            if len(trace2_points) > 1:
                new_path = DashedVMobject(Line(trace2_points[-2], trace2_points[-1]), num_dashes=1, equal_lengths=True)
                trace2.add(new_path)

        trace.add_updater(update_trace)
        trace2.add_updater(update_trace2)

        # Add elements to the scene
        self.play(Write(axes), Write(labels))
        self.add(d, d2, trace, trace2)
        self.wait(2)
        self.play(d.animate.move_to([0, 3, 0]), d2.animate.move_to([0, 2, 0]))
        self.wait(2)
        self.play(d.animate.move_to([0, -3, 0]), d2.animate.move_to([0, -2, 0]))
        self.wait(2)
        trace.clear_updaters()
        trace2.clear_updaters()
```

**Video:**

https://github.com/360macky/generative-manim/assets/47821093/54bb2379-ace8-4e9d-a8df-694ec4b7b8be

**Comment by LucaM185** [comment]
I might know what's happening here, i tried building exactly that last year... 
i had the problem that the first scene rendered had a name and the next ones kept the name of the first (for some dumb reason this is the default behaviour of manim) i fixed it by setting     

force_output_as_scene_name = True 

in manim. i literally ctrl-clicked my way from Scene->CairoRenderer->SceneFileWriter, ctrl-f "force" and you should find it. 
it requires some changes to the structure of the code. 

i didn't look at your code too closely, but having built this before, it seems like that kind of issue, since you are likely running in docker you will have to find a smarter way to do this...

**Comment by 360macky** [comment]
> I might know what's happening here, i tried building exactly that last year... i had the problem that the first scene rendered had a name and the next ones kept the name of the first (for some dumb reason this is the default behaviour of manim) i fixed it by setting
> 
> force_output_as_scene_name = True
> 
> in manim. i literally ctrl-clicked my way from Scene->CairoRenderer->SceneFileWriter, ctrl-f "force" and you should find it. it requires some changes to the structure of the code.
> 
> i didn't look at your code too closely, but having built this before, it seems like that kind of issue, since you are likely running in docker you will have to find a smarter way to do this...

Hi @LucaM185!, thank you so much!, I've thought about rewriting the scene name to make it different before.
I didn't think Manim would have this option, but I guess it has to do with speeding up the rendering process and not re-rendering already made scenes (I think).

**Comment by LucaM185** [comment]
Yes, if you reuse the same scene name it intelligently reuses previous partial_movie_files if they have the same hash. This is obviously not useful if you have completely different scenes, so I don't see a performance penalty in using different names... Just make sure you delete the folders after a while because of storage costs

**Comment by 360macky** [comment]
> Yes, if you reuse the same scene name it intelligently reuses previous partial_movie_files if they have the same hash. This is obviously not useful if you have completely different scenes, so I don't see a performance penalty in using different names... Just make sure you delete the folders after a while because of storage costs

Great, thanks a lot @LucaM185, we will start implementing the `force_output_as_scene_name = True` instruction [in the following days](https://github.com/360macky/generative-manim/pull/26).
